### PR TITLE
[RW-9520][risk=low] Added error modal for creating apps

### PR DESCRIPTION
**Pull request by evrii**
Description:
Added an error modal for when there is a problem creating a Cromwell environment.

In order to test, create an RStudio environment, and then create a Cromwell environment. You should see something like this:
image

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: ui/src/app/components/apps-panel/expanded-app.tsx [hunk 1]
```diff
@@ -147,7 +147,9 @@ const RStudioButtonRow = (props: {
   const onClickCreate = withErrorModal(
     {
       title: 'Error Creating RStudio Environment',
-      message: 'Please refresh the page.',
+      message:
+        'Please wait a few minutes and try to create your RStudio Environment again.',
```
Comment (evrii) [inline ui/src/app/components/apps-panel/expanded-app.tsx:151]: Updated to match wording of Cromwell Error modal.

File: ui/src/app/components/cromwell-configuration-panel.tsx [hunk 2]
```diff
@@ -62,12 +63,25 @@ const PanelMain = fp.flow(
...
       if (!creating) {
         setCreating(true);
-        appsApi().createApp(workspace.namespace, defaultCromwellConfig);
-        onClose();
-        setTimeout(() => setSidebarActiveIconStore.next('apps'), 3000);
+        appsApi()
+          .createApp(workspace.namespace, defaultCromwellConfig)
+          .then(() => onDismiss())
+          .catch(() =>
+            notificationStore.set({
```
Comment (evrii) [inline ui/src/app/components/cromwell-configuration-panel.tsx:78]: Used `notificationStore.set({...})` instead of `withErrorModal`, because withErrorModal does not currently return a promise. I am willing to update `withErrorModal` to return a promise, if that is the preferred approach.

**Review by evrii — COMMENTED** [review]
I did not add a test for this, because I think that this would be a great candidate for a contract test. The contract tests are currently blocked, but I can try to add it after the fact.

File: ui/src/app/components/cromwell-configuration-panel.spec.tsx [hunk 3]
```diff
@@ -101,19 +101,26 @@ describe('CromwellConfigurationPanel', () => {
...
 
+    await waitOneTickAndUpdate(wrapper);
+
     expect(spyCreateApp).toHaveBeenCalledTimes(1);
     expect(spyCreateApp).toHaveBeenCalledWith(
       WorkspaceStubVariables.DEFAULT_WORKSPACE_NS,
       defaultCromwellConfig
     );
-    expect(onClose).toHaveBeenCalledTimes(1);
+    await expect(onClose).toHaveBeenCalledTimes(1);
```
Comment (jmthibault79) [inline ui/src/app/components/cromwell-configuration-panel.spec.tsx:124]: are you sure this is necessary?

File: ui/src/app/components/cromwell-configuration-panel.spec.tsx [hunk 4]
```diff
@@ -101,19 +101,26 @@ describe('CromwellConfigurationPanel', () => {
   it('start button should create cromwell and close panel', async () => {
     jest
       .spyOn(appsApi(), 'listAppsInWorkspace')
-      .mockImplementationOnce(() => Promise.resolve([]));
+      .mockImplementation((): Promise<any> => Promise.resolve([]));
```
Comment (jmthibault79) [inline ui/src/app/components/cromwell-configuration-panel.spec.tsx:104]: why this change?

File: ui/src/app/components/cromwell-configuration-panel.tsx (hunk 2 above)
Comment (jmthibault79) [reply ui/src/app/components/cromwell-configuration-panel.tsx]: (resolving because outdated)

**Review by jmthibault79 — APPROVED** [review]
Tested locally: RStudio then Cromwell, and Cromwell then RStudio.

File: ui/src/app/components/cromwell-configuration-panel.spec.tsx (hunk 4 above)
Comment (evrii) [reply ui/src/app/components/cromwell-configuration-panel.spec.tsx]: This was accidental. Thanks for bringing it to my attention.
